### PR TITLE
gcc/analyzer: fix false positives for NULL

### LINKED ISSUE #13158: [CWE-690]
Hello everyone :smiley: Tried to compile neovim with gcc10 and flag `-fanalyzer` and found some [CWE-690]  NULL dereference. Here's a log, but I recommend trying to compile with `-fanalyzer` to see more. :smile: 
[log.txt](https://github.com/neovim/neovim/files/5435319/logg.txt)

### PR DESCRIPTION
**Pull request by janlazo**
Close https://github.com/neovim/neovim/issues/13158

cc @fiplox

### CODE REVIEW

**Comment by janlazo** [comment]
`src/nvim/testdir/test_pyx2.vim` crashed in Appveyor. https://ci.appveyor.com/project/neovim/neovim/builds/36192142/job/oyapk15f87aj9yx7

**Comment by janlazo** [comment]
Appveyor passed now. Random failure in previous build?

**Comment by janlazo** [comment]
Will merge if Appveyor does not fail on the same test.

File: src/nvim/api/vim.c [hunk 1]
```diff
@@ -2687,6 +2687,9 @@ void nvim__screenshot(String path)
 
 static void clear_provider(DecorProvider *p)
 {
+  if (p == NULL) {
```
Comment (bfredl) [inline src/nvim/api/vim.c:2690]: hmm this should rather have the `NONNULL` attribute (I can fix it next time I touch this function which will be very soon).

**Pull request merged**